api/src/main/java/org/apache/iceberg/util/BinaryUtil.java

```diff
@@ -28,12 +28,23 @@
   private BinaryUtil() {
   }
 
+  private static final ByteBuffer EMPTY_BYTE_BUFFER = ByteBuffer.allocate(0);
```

Comment:
I can make this lazily instantiated if we care. Or just return a new empty byte buffer instance every time in the function, but seems unnecessary to have all of those allocations.